### PR TITLE
fix bugs-metrics dashboard and switching mode, and also enable scro…

### PR DESCRIPTION
**Pull request by kjcho-msft**
Fixing bugs-metrics dashboard spinning issue and switching mode crashing issue
Enabling scroll for Jobs page

### CODE REVIEW

File: Website/Packages/datax-jobs/src/modules/jobs/components/sparkJobs.jsx [hunk 1]
```diff
@@ -134,6 +134,12 @@ const mapStateToProps = state => ({});
...
 // Styles
+const rootStyle = {
+    display: 'flex',
+    flexDirection: 'row',
+    overflowY: 'auto'
+};
+
 const contentStyle = {
     paddingLeft: 30,
     paddingRight: 30,
```
[the remark this answers is not in the document]
Comment (kjcho-msft) [reply Website/Packages/datax-jobs/src/modules/jobs/components/sparkJobs.jsx]: Panel is still used in render() and this is a style for its content.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/input/inputSettingsContent.jsx [hunk 2]
```diff
@@ -65,7 +65,7 @@ export default class InputSettingsContent extends React.Component {
     renderLeftPane() {
         if (this.props.input.mode === Models.inputModeEnum.batching) {
             let batchData = undefined;
-            if (this.props.batchInputs != undefined && this.props.selectedFlowBatchInputIndex != undefined) {
```
[the remark this answers is not in the document]
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/input/inputSettingsContent.jsx]: Added an item to backlog.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/input/inputSettingsContent.jsx [hunk 3]
```diff
@@ -65,7 +65,7 @@ export default class InputSettingsContent extends React.Component {
...
+            if (this.props.batchInputs !== null && this.props.selectedFlowBatchInputIndex !== undefined) {
```
[the remark this answers is not in the document]
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/input/inputSettingsContent.jsx]: Updated. Thanks

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowHelpers.js [hunk 4]
```diff
@@ -441,6 +441,7 @@ export function convertFlowToConfig(flow, query) {
     let outputTemplates = [...flow.outputTemplates];
     let rules = [...flow.rules];
     let batchList = flow.batchList ? [...flow.batchList] : [];
+    let batchInputs = flow.batchInputs !== null && flow.batchInputs ? [...flow.batchInputs] : [Models.getDefaultBatchInputSettings()];
```
[the remark this answers is not in the document]
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/flowHelpers.js]: Updated.

[the remark this answers is not in the document]
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/flowHelpers.js]: Updated.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/input/inputSettingsContent.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/input/inputSettingsContent.jsx]: Updated. Thanks

File: Website/Packages/datax-metrics/src/modules/metrics/components/metricExplorer.jsx [hunk 5]
```diff
@@ -129,8 +129,7 @@ const contentStyle = {
     width: '100%',
     display: 'flex',
     flexDirection: 'column',
-    overflowY: 'auto',
-    paddingBottom: 80
```
Comment (carlbrochu) [inline Website/Packages/datax-metrics/src/modules/metrics/components/metricExplorer.jsx]: i think we want to keep 80 and not 20 tho.  Is the effect the same?

File: Services/DataX.Flow/DataX.Flow.DeleteHelper/ConfigDeleter.cs [hunk 6]
```diff
@@ -99,8 +99,8 @@ public async Task<ApiResult> DeleteFlow(JObject jObject)
             _inputEventhubConnectionStringRef = Helper.IsKeyVault(diag.EventhubConnectionString) ? diag.EventhubConnectionString : Helper.GenerateNewSecret(_keySecretList, _engineEnvironment.EngineFlowConfig.SparkKeyVaultName, ConfigName + "-input-eventhubconnectionstring", _engineEnvironment.EngineFlowConfig.SparkType, diag.EventhubConnectionString, false);
             diag.EventhubConnectionString = _inputEventhubConnectionStringRef;
 
-            if (diag.InputType == Constants.InputType_EventHub)
```
Comment (carlbrochu) [inline Services/DataX.Flow/DataX.Flow.DeleteHelper/ConfigDeleter.cs]: Can this be unit tested?

File: Website/Packages/datax-metrics/src/modules/metrics/components/metricExplorer.jsx (hunk 5 above)
Comment (kjcho-msft) [reply Website/Packages/datax-metrics/src/modules/metrics/components/metricExplorer.jsx]: Yes I don't see any difference. And just for consistency, I decided to keep 20.

File: Services/DataX.Flow/DataX.Flow.DeleteHelper/ConfigDeleter.cs (hunk 6 above)
Comment (kjcho-msft) [reply Services/DataX.Flow/DataX.Flow.DeleteHelper/ConfigDeleter.cs]: Added some unit tests for the helper functions in ConfigDeleter. For E2E, we should use scenario tester.

**Pull request merged**